### ISSUE TITLE
Bundling CJS module breaks strict mode

### ISSUE THREAD

**Issue by saeris-sanoora** [systemjs/builder#387]
Suppose you have a CommonJS-style module called `app.js` which has a `"use strict";` pragma at the top to enable strict mode:

```js
"use strict";
var something = require("something");
// ... etc.
```

Bundling this module will insert additional lines prior to the `"use strict";` pragma:

```js
System.registerDynamic("app/app.js", ["something"], true, function(req, exports, module) {
  ;
  var global = this,
      __define = global.define;
  global.define = undefined;
  ;
  "use strict";
  var something = req("something");
  // ... etc.
  global.define = __define;
  return module.exports;
});
```

This breaks the pragma and disables strict mode, since the pragma must be the first expression in the function block.

Currently, I am working around this problem by inserting a build step prior to bundling which just wraps every JS file in an IIFE. I don't think I'd recommend that as the official fix, though. It might be simpler to just check whether the module source matches `/^['"]use strict['"]/` and, if so, insert an identical expression prior to the `global`/`define` boilerplate header.

Thoughts?

**Comment by dristic** [comment]
This definitely makes sense :+1: 

Just adding some reference here, according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode the `"use strict"` modifier has to be the _first_ statement in the function. Parsing out the use strict sounds like a good approach as a solution.

**Comment by dristic** [comment]
I dropped a PR that could potentially fix the issue, but definitely needs some feedback and review.

**Comment by guybedford** [comment]
Released in 0.14.11.
